packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.classNames.ts

```diff
@@ -0,0 +1,97 @@
+import { memoizeFunction } from '../../Utilities';
+import { mergeStyles, IStyle } from '../../Styling';
+import { ISpinButtonStyles } from './SpinButton.Props';
+import { KeyboardSpinDirection } from './SpinButton';
+import { Position } from '../../utilities/positioning';
+
+export interface ISpinButtonClassNames {
+  root: string;
+  labelWrapper: string;
+  icon: string;
+  label: string;
+  spinButtonWrapper: string;
+  input: string;
+  arrowBox: string;
+}
+
+export const getClassNames = memoizeFunction((
+  styles: ISpinButtonStyles,
+  disabled: boolean,
+  isFocused: boolean,
+  keyboardSpinDirection: KeyboardSpinDirection,
+  labelPosition: Position = Position.start
+): ISpinButtonClassNames => {
+  return {
+    root: mergeStyles(
+      styles.root
+    ) as string,
+    labelWrapper: mergeStyles(
+      styles.labelWrapper,
+      _getStyleForLabelBasedOnPosition(labelPosition, styles)
+    ) as string,
+    icon: mergeStyles(
+      styles.icon,
+    ) as string,
+    label: mergeStyles(
+      styles.label
+    ) as string,
+    spinButtonWrapper: mergeStyles(
+      styles.spinButtonWrapper,
+      _getStyleForRootBasedOnPosition(labelPosition, styles),
+      !disabled && [
+        {
+          ':hover': styles.spinButtonWrapperHovered
+        },
+        isFocused && {
+          // This is to increase the specifity of the focus styles
+          // and make it equal to that of the hover styles.

```

Comment:
You want greater, not equal, specificity, since you can't insure the order the styles will go onto the page.